analyses/molecular-subtyping-embryonal/00-embryonal-select-pathology-dx.Rmd

```diff
@@ -0,0 +1,171 @@
+      pathology_diagnosis == "Supratentorial or Spinal Cord PNET"
+  ) %>%
+  count(pathology_diagnosis) %>%
+  arrange(desc(n))
+```
+
+For the most part, this is as we would expect given the 2016 WHO classifications.
+There is an ependymoma sample and a high-grade glioma sample that should probably be excluded.
+
+Let's take a look at the free text field.
+
+```{r}
+histologies_df %>% 
+  filter(
+    str_detect(str_to_lower(pathology_diagnosis), "neuroblastoma") |
+      pathology_diagnosis == "Supratentorial or Spinal Cord PNET"
+  ) %>%
+  group_by(pathology_free_text_diagnosis) %>%
+  tally() %>%
+  arrange(desc(n))
+```
+
+No medulloblastoma or ATRT samples here.
+
+## Pathology diagnosis strings for inclusion
+
+These are the terms that we'll collapse together with `|` to detect strings in the `pathology_diagnosis` column.
+ 
+```{r}
+path_dx_terms <- c(
+  "supratentorial or spinal cord pnet"
+) # "neuroblastoma" will also need to be detect in the `pathology_diagnosis` column, but the logic around it is a bit more complex so we will handle this term specifically when we filter
+```
+
+These are the terms that we'll collapse together with `|` to detect strings in the `pathology_free_text_diagnosis` column.
+
+```{r}
+free_text_dx_terms <- c(
+  "embryonal tumor with multilayer rosettes, ros (who grade iv)",
+  "embryonal tumor, nos, congenital type",
+  "ependymoblastoma",
+  "medullooepithelioma"
+)
+```
+
+Let's take a look at a first attempt using these terms as described above.
+
+```{r}
+filtered_on_dx_df <- histologies_df %>%
+  filter(str_detect(str_to_lower(pathology_diagnosis), 
+                    paste0(path_dx_terms, collapse = "|")) |
+           pathology_diagnosis == "Other" &&
+           str_detect(str_to_lower(pathology_free_text_diagnosis), 
+                      paste0(free_text_dx_terms, collapse = "|")) |
+           (str_detect(str_to_lower(pathology_diagnosis), "neuroblastoma") & !(str_detect(str_to_lower(primary_site), "other locations nos")) & !(str_detect(str_to_lower(pathology_free_text_diagnosis), "peripheral")))) %>%
+  select(Kids_First_Biospecimen_ID, 
+         sample_id, 
+         Kids_First_Participant_ID,
+         pathology_diagnosis,
+         pathology_free_text_diagnosis,
+         integrated_diagnosis,
+         primary_site,
+         broad_histology)
+filtered_on_dx_df
+```
+
+Let's tally the values in `pathology_diagnosis` in this data frame.
+
+```{r}
+filtered_on_dx_df %>%
+  count(pathology_diagnosis) %>%
+  arrange(desc(n))
+```
+
+We know we need to drop medulloblastoma and ATRT samples.
+We can also exclude LGG and neurofibroma samples on the basis of the values in `pathology_diagnosis`.
+
+```{r}
+exclude_path_dx_terms <- c("medulloblastoma",
+                           "atypical teratoid rhabdoid tumor (atrt)",
+                           "low-grade glioma/astrocytoma",
+                           "neurofibroma/plexiform;other")
+```
```

Comment:
_This is a more general comment that could have gone in several places in line._

My interpretation of https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/752#issuecomment-697000066 is that we can do the following:

1. Exact matching for `Supratentorial or Spinal Cord PNET`, e.g., `pathology_diagnosis  == "Supratentorial or Spinal Cord PNET"` - but use `%in%` for future proofing (more on that below). 
2. If `pathology_diagnosis == "Other"`, then detection in `pathology_free_text_diagnosis` of
```
embryonal tumor with multilayer rosettes, ros (who grade iv)
embryonal tumor, nos, congenital type
ependymoblastoma
medullooepithelioma
```
(`str_detect()`, `str_to_lower()`, `paste0(..., collapse = "|")` all working together like we have)

3. Detection of `neuroblastoma` in `pathology_diagnosis` (the `str_detect()`, etc. strategy again ☝️ ) and `primary_site != "Other locations NOS"` and `pathology_free_text_diagnosis` does not contain `"peripheral"` (`str_detect(str_to_lower(...), negate = TRUE)`) (Side note: Do we want `&` or `&&` throughout? I have not given much thought about how `&&` behaves with filtering with `dplyr`.)

Which I think means we will not need to do any exclusion on the basis of terms that indicate that a specimen is medulloblastoma or ATRT. 

Instead you can have a set of terms for exact matching in the JSON file that you use in conjunction with `%in%` to filter to exact matches and then the free text terms for inclusion. Here's an example in the MB module for putting together a JSON file to be used in this way: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/a0ae052487a72790f7eaec318815f47eae13b466/analyses/molecular-subtyping-MB/00-mb-select-pathology-dx.Rmd#L106

And here's the filtering: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/a0ae052487a72790f7eaec318815f47eae13b466/analyses/molecular-subtyping-MB/01-filter-and-batch-correction.R#L53

I agree with your choice to keep the NBL steps separate/just in the filtering.